### PR TITLE
Fix formfield ts

### LINKED ISSUE #5118: Grommet global theme for formField complains that 'side' is incompatible (TypeScript)
### Expected Behavior

The theme prop added to the `Grommet` component should recognize the key `side` in `BorderType` when configuration global theme for formField

### Actual Behavior

I am trying to set a global theme for my formField where the border is displayed all around the TextInput. This is my theme configuration, which I took from the story book found here: https://storybook.grommet.io/?path=/story/input-form-field-custom-states--field-custom-states

```TypeScript
const theme = {
  formField:{
    border: {
      error: {
        color: 'border'
      },
      color: 'border',
      side: 'all'
    },
    round: '4px'
  }
}
```

The theme prop that is passed to the `Grommet` component complains that `side` is incompatible. I get this message
image

### Steps to Reproduce

1. 2. 3.

### Your Environment

- Grommet version: 2.17.1
- Browser Name and version:
- Operating System and version (desktop or mobile):

### PR DESCRIPTION
**Pull request by britt6612**
#### What does this PR do?
This PR adds `formfield` theme to the base.d.ts for typescript
#### Where should the reviewer start?
theme/base.d.ts
#### What testing has been done on this PR?
manually 

```
const theme = {
  formField:{
    border: {
      error: {
        color: 'border'
      },
      color: 'border',
      side: 'all'
    },
    round: '4px'
  }
}

```
#### How should this be manually tested?
storybook with a ts story
#### Any background context you want to provide?
previously there was `formfield.border` which was set to `bordertype` however this does not include colors. This is why if someone was to use  `formfield.border.color` it would throw an error since `color` string is not  included in the `bordertype`
Also found that `formfield.extend` was missing so I added it in.
#### What are the relevant issues?
closes #5118 
#### Do the grommet docs need to be updated?
no
#### Should this PR be mentioned in the release notes?
no
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

File: src/js/themes/base.d.ts [hunk 1]
```diff
@@ -726,7 +726,14 @@ export interface ThemeType {
...
   formField?: {
-    border?: BorderType;
+    border?: {
+      color?: ColorType;
+      position?: BorderType;
+      side?: BorderType;
+      error?: {
+        color?: ColorType;
+      };
+    };
```
Comment (ShimiSun) [inline src/js/themes/base.d.ts:736]: Wouldn't it be a better practice to extend the `BroderType` types instead of hard coding the prop values?

Comment (britt6612) [reply src/js/themes/base.d.ts:736]: I was inclined to do this but saw other areas that used `border.color` so I can add colors to extend `BorderType` 
Would you want me to change all of the other `border` areas to  be `BorderType` ?

Comment (taysea) [reply src/js/themes/base.d.ts:736]: I think what can be done instead is such:
```
border?: BorderType;
```
BorderType already includes the necessary typings for color, position, etc

Comment (britt6612) [reply src/js/themes/base.d.ts]: Your right i was making it to complicated 😓 I just added the `error.color` to the  `border?:`

File: src/js/themes/base.d.ts [hunk 2]
```diff
@@ -726,7 +726,11 @@ export interface ThemeType {
     round?: RoundType;
   };
   formField?: {
-    border?: BorderType;
+    border?: BorderType & {
+      error?: {
+        color?: ColorType;
```
Comment (taysea) [inline src/js/themes/base.d.ts:731]: Is there anything else that would be on border.error? Or it's just color that we can change

Comment (britt6612) [reply src/js/themes/base.d.ts:731]: https://v2.grommet.io/formfield#formField.border.error.color 
Thats it according to the docs

Comment (britt6612) [reply src/js/themes/base.d.ts:731]: also checked code:
https://github.com/grommet/grommet/blob/master/src/js/components/FormField/FormField.js#L264

**Review by taysea — APPROVED** [review]
LGTM!

File: src/js/themes/base.d.ts (hunk 2 above)
Comment (taysea) [reply src/js/themes/base.d.ts:731]: Great thanks!

**Review by ShimiSun — APPROVED** [review]

**Pull request merged**